File: Python/Data Structures/Linked List/Merge-Two-Sorted-Lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
        if l1 is None and l2 is None:
            return l1
        if l1 is None and l2:
            return l2
        if l2 is None and l1:
            return l1
             
        l = list()
        while l1 or l2:
            if l1:
                l.append(l1.val)
                l1 = l1.next
            if l2:
                l.append(l2.val)
                l2 = l2.next
                
        dummy=ListNode()
        tail = dummy
        for i in sorted(l):
            tail.next = ListNode(i)
            tail = tail.next                
        
        return dummy.next   

File: Golang/Merge-Two-Sorted-Lists.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
// import (
    // "sort"
    // "fmt"
// )
func mergeTwoLists(list1 *ListNode, list2 *ListNode) *ListNode {
    
    current := list1
    vals := []int{}
    
    for current != nil {
        vals = append(vals, current.Val)
        current = current.Next
    }
    
    current = list2
    
    for current != nil {
        vals = append(vals, current.Val)
        current = current.Next
    }
    
    sort.Slice(vals, func(i, j int) bool {
        return vals[i]<vals[j]
    })
    
    dummy := ListNode{0, nil}
    tail := &dummy
    
    for _, val := range vals {
        tail.Next = &ListNode{val, nil}
        tail = tail.Next
    }

    return dummy.Next
}
